components/ai_chat/content/browser/page_content_fetcher.cc

```diff
@@ -431,4 +444,17 @@ void FetchPageContent(content::WebContents* web_contents,
   fetcher->Start(std::move(extractor), invalidation_token, std::move(callback));
 }
 
+void GetSearchSummarizerKey(
+    content::WebContents* web_contents,
+    mojom::PageContentExtractor::GetSearchSummarizerKeyCallback callback) {
+  auto* primary_rfh = web_contents->GetPrimaryMainFrame();
+  DCHECK(primary_rfh->IsRenderFrameLive());
+
+  auto* fetcher = new PageContentFetcher(nullptr);
```

Comment:
Wow, struggled to understand the lifetime of this class a bit - it basically ties its lifetime to whatever task it's trying to do, and deletes itself when it's done or failed? :+1:

Reply:
I'm just following the current implementation here. :)